mllib/src/main/scala/org/apache/spark/ml/stat/ChiSquareTest.scala

```diff
@@ -63,40 +55,47 @@ object ChiSquareTest {
    */
   @Since("2.2.0")
   def test(dataset: DataFrame, featuresCol: String, labelCol: String): DataFrame = {
-    val spark = dataset.sparkSession
-    import spark.implicits._
-
-    SchemaUtils.checkColumnType(dataset.schema, featuresCol, new VectorUDT)
-    SchemaUtils.checkNumericType(dataset.schema, labelCol)
-    val rdd = dataset.select(col(labelCol).cast("double"), col(featuresCol)).as[(Double, Vector)]
-      .rdd.map { case (label, features) => OldLabeledPoint(label, OldVectors.fromML(features)) }
-    val testResults = OldStatistics.chiSqTest(rdd)
-    val pValues = Vectors.dense(testResults.map(_.pValue))
-    val degreesOfFreedom = testResults.map(_.degreesOfFreedom)
-    val statistics = Vectors.dense(testResults.map(_.statistic))
-    spark.createDataFrame(Seq(ChiSquareResult(pValues, degreesOfFreedom, statistics)))
+    test(dataset, featuresCol, labelCol, false)
   }
 
   /**
    * @param dataset  DataFrame of categorical labels and categorical features.
    *                 Real-valued features will be treated as categorical for each distinct value.
    * @param featuresCol  Name of features column in dataset, of type `Vector` (`VectorUDT`)
    * @param labelCol  Name of label column in dataset, of any numerical type
-   * @return Array containing the SelectionTestResult for every feature against the label.
+   * @param flatten  If false, the returned DataFrame contains only a single Row, otherwise, one
```

Comment:
Is the behavior identical to the existing method if false?
Hm OK this is a little odd API-wise but I understand you're trying to keep compatibility while adding a better representation too. I'm neutral on it.

The existing method below seems like it had an OK purpose too, to return the results as objects rather than a DataFrame, but I also don't feel strongly about retaining it.

Reply:
> Is the behavior identical to the existing method if false?

Yes, since all methods share the same logic to compute test results, the differences lie in how to represent the results.

> The existing method below seems like it had an OK purpose too, to return the results as objects rather than a DataFrame

I am neutral on keeping it or not. How do you @huaxingao think about it? I can add it back it you want it.

Reply:
I am OK with the change.